iroh-net/src/magicsock.rs

```diff
@@ -1767,10 +1767,22 @@ impl Actor {
                 discovery_events = events;
             }
         }
+
+        let mut receiver_closed = false;
+        let mut portmap_watcher_closed = false;
+        let mut link_change_closed = false;
         loop {
             inc!(Metrics, actor_tick_main);
             tokio::select! {
-                Some(msg) = self.msg_receiver.recv() => {
+                msg = self.msg_receiver.recv(), if !receiver_closed => {
+                    let Some(msg) = msg else {
+                        trace!("tick: magicsock receiver closed");
+                        inc!(Metrics, actor_tick_other);
+
+                        receiver_closed = true;
+                        continue;
+                    };
+
```

Comment:
This means there are no senders anymore.  So either this is fatal or new senders can be created I think.  If new senders can be created this needs to be polled again, so the simple `receiver_closed` boolean is not sufficient because it always disables this branch.

Reply:
No new senders can be created at that point (it's an mpsc channel so there'd have to be an existing sender to clone) so yeah I guess it'd be fatal

Reply:
Thanks for checking!